node-red-project/flows.json

```diff
@@ -2591,7 +2591,7 @@
         "z": "9269520ddb532b00",
         "g": "b334a2375eb8cb42",
         "name": "setup telemetry payload & URL",
-        "func": "// format the payload with a timestamp\nlet values = { ...msg.payload }\nmsg.payload = { \n    ts: Date.now(),\n    values: values\n    }\n\n// The server URL, port, and device access token must be stored in the config context\n// before this will work\nlet serverUrl = global.get(\"serverUrl\", \"config\")\nlet serverPort = global.get(\"serverPort\", \"config\")\nlet accessToken = global.get(\"accessToken\", \"config\")\nmsg.url = `${serverUrl}:${serverPort}/api/v1/${accessToken}/telemetry`\nmsg.method = \"POST\";\nreturn msg;",
+        "func": "// format the payload with a timestamp\nlet values = {\n    ...msg.payload,\n    deviceName: global.get(\"deviceName\", \"config\"),\n    location: global.get(\"location\", \"config\"),\n    latitude: global.get(\"latitude\", \"config\"),\n    longtitude: global.get(\"longtitude\", \"config\")\n    }\nmsg.payload = { \n    ts: values.frame_time,\n    values: values\n    }\n\n// The server URL, port, and device access token must be stored in the config context\n// before this will work\nlet serverUrl = global.get(\"serverUrl\", \"config\")\nlet serverPort = global.get(\"serverPort\", \"config\")\nlet accessToken = global.get(\"accessToken\", \"config\")\nmsg.url = `${serverUrl}:${serverPort}/api/v1/${accessToken}/telemetry`\nmsg.method = \"POST\";\nreturn msg;",
```

Comment:
Don't put the location telemetry in this message. This will update lat/long on every event, which would be unnecessarily too often. These only need to change when the "deployment" and/or "location" change or maybe even on device restart.

Reply:
Let's talk about that problem in a new branch, just leave them off for now.

Reply:
ok. Though I don't understand why including them would update anything if they are unchanged.

Reply:
Just to follow up, based on our convo: Don't want to include [slow changing dimensions](https://en.wikipedia.org/wiki/Slowly_changing_dimension) in the transactional (event) data.  We also want to preserve the history of location changes in it's own location, not attached to event.  There MAY be a good use case for including a foreign key or referential integrity with events, but we can do handle that with time-based relationship too, since location and events have a 1:many relationship.